reana_workflow_controller/rest/workflows.py

```
@@ -888,3 +891,258 @@ def get_workflow_retention_rules(workflow_id_or_name: str, user: str):
+          type: string
+        - name: workflow_id_or_name
+          in: path
+          description: Required. Analysis UUID or name.
+          required: true
+          type: string
+        - name: user_email_to_share_with
+          in: query
+          description: >-
+            Required. User to share the workflow with.
+          required: true
+          type: string
+        - name: message
+          in: query
+          description: Optional. Message to include when sharing the workflow.
+          required: false
+          type: string
+        - name: valid_until
+          in: query
+          description: Optional. Date when access to the workflow will expire (format YYYY-MM-DD).
+          required: false
+          type: string
+      responses:
+        200:
+          description: >-
+            Request succeeded. The workflow has been shared with the user.
+          schema:
+            type: object
+            properties:
+              message:
+                type: string
+              workflow_id:
+                type: string
+              workflow_name:
+                type: string
+          examples:
+            application/json:
+              {
+                "message": "The workflow has been shared with the user.",
+                "workflow_id": "cdcf48b1-c2f3-4693-8230-b066e088c6ac",
+                "workflow_name": "mytest.1"
+              }
+        400:
+          description: >-
+            Request failed. The incoming data specification seems malformed.
+          schema:
+            type: object
+            properties:
+              message:
+                type: string
+              errors:
+                type: array
+                items:
+                  type: string
+          examples:
+            application/json:
+              {
+                "message": "Malformed request.",
+                "errors": ["Missing data for required field."]
+              }
+        403:
```

I think the 403 error is never returned?

---

That is correct, nicely spotted. I removed it. In the future once this is implemented (https://github.com/reanahub/reana-server/issues/663) it could be readded.

The change has been applied to this superseding PR: #552